modules/standard/Math.chpl

```diff
@@ -1106,56 +1112,65 @@ module Math {
   }
 
 
-  /* Returns the Bessel function of the second kind of order `0` of `x`,
-     if and only if the value of `x` is greater than 0*/
-  extern "y0f" proc y0(x: real(32)): real(32);
+  /* Returns the Bessel function of the second kind of order `0` of `x`, where
+     `x` must be greater than 0 */
+  inline proc y0(x: real(32)): real(32) {
+    if boundsChecking && x < 0 then
+      HaltWrappers.boundsCheckHalt("Input value for y0() must be greater than 0");
```

Comment:
Good catch -- force pushed to use `<= 0`

Reply:
Hmm, so the yn.chpl test runs with x=0 and checks that the result is -INF: 

https://github.com/chapel-lang/chapel/blob/master/test/library/standard/Math/yn.chpl#L34-L35

"""
y0 returns the Bessel function of the second kind of order 0 of x. It may signal underflow if x is too large. If x is negative, y0 signals a domain error; if it is zero, y0 signals overflow and returns -infin;.
"""

Just update that test to avoid that case, or go back to < 0, and update the error message?

Reply:
Ah, sorry.  Well, the die.net man page says it's an error, but then also says it must return `-HUGE_VAL` which for IEEE floating point is `-INF`.  That means we could go ahead and allow a zero and depend on the `-INF` happening.

So, I guess I got it backwards.  The `if` test should be `x < 0` and the error message should say greater than or equal to zero.